### ISSUE TITLE
Building 2.6.1 on Windows

### ISSUE THREAD

**Issue by jariq** [opendnssec/SoftHSMv2#547]
Hello volunteers,

I'm trying to understand how to build SoftHSM 2.6.1 on Windows platform and I would be grateful if someone could shed some light into my suffering by helping me to find the answers to the following questions:

1. It seems that by removing ` win32/Configure.py` script in https://github.com/opendnssec/SoftHSMv2/commit/f0e3afe77c7019a81cea7c4622031f5403c00859 CMake was promoted to be the only build system for Windows used in SoftHSMv2 project. Is this observation correct? Is CMake the only option now? If yes, was this change intentional?

    1.1 Why wasn't [WIN32-NOTES.md](https://github.com/opendnssec/SoftHSMv2/blob/2.6.1/WIN32-NOTES.md) document updated? It seem to be providing incorrect information now that `win32/Configure.py` is not present in the project anymore.

    1.2 Why wasn't the whole [`win32`](https://github.com/opendnssec/SoftHSMv2/tree/2.6.1/win32) directory removed too? I may be wrong but it does not seem to be used anymore now that  `win32/Configure.py` is gone.

3. [AppVeyor build for 2.6.1](https://ci.appveyor.com/project/opendnssec/softhsmv2/builds/32541582) failed with red color. Even [AppVeyor build for 2.6.0](https://ci.appveyor.com/project/opendnssec/softhsmv2/builds/31526581) failed regardless of green color (inspect [console output](https://ci.appveyor.com/project/opendnssec/softhsmv2/builds/31526581/job/w5954byeflwcir8u#L26579) in case of any doubt). Was SoftHSM 2.6.1 built successfully on any Windows machine before the release?

I'm currently trying to get 2.6.1 building locally on my computer but the process is full of undocumented mostly CMake related traps. I believe that finding the answers to the questions above would help me a lot in this effort.

Thanks, Jaroslav

**Comment by rijswijk** [comment]
We're looking into this, it may take some days before we can respond.

**Comment by halderen** [comment]
Yes, using CMake was much simpler than the configure python script, which would be possible with updated Botan dependencies, and did fail on occasions.
Regarding updating WIN32-NOTES.md and removal of several unused files, that remains to be done.  The main reason is that the tests themselves don't all run on windows.  So I can build and verify it is more-or-less functional, but not fully test.  I do not have access to a windows build environment, making the development cycle slow as I rely on AppVeyor.  For that same reason the 2.6.0 was green (it did not run all tests), while 2.6.1 runs tests.  Althrough the build is (or should be) complete, a proper CI for windows isn't completed.   Given the changes from 2.6.0 to 2.6.1, I did not pay much attention to the windows build.

**Comment by jariq** [comment]
@rijswijk is there any update on this issue?

**Comment by halderen** [comment]
None yet, I expect this is something I'd might pick up during the vacation period, at least the coming three weeks are booked.

**Comment by bjosv** [comment]
FYI: Building the branch `develop` on Windows seems to work fine currently, but needs a minor fix to build with tests (see #733).

**Comment by FabianWirtz** [comment]
@halderen Is it possible to create a release with a working windows build?
